08-transport.Rmd

```diff
@@ -300,14 +322,16 @@ Desire lines contain only two vertices (their beginning and end points) but rout
 
 Routes are generated from desire lines --- or more commonly origin-destination pairs --- using routing algorithms.
 Routing algorithms can either run locally or remotely.
-Local routing is computationally intensive because requires having a representation of the route network stored on your computer (we will see how in section \@ref(route-networks)).
-Remote routing services use an API to receive queries about origin and destination points, and then return the results over the internet.
+Routing is computationally intensive and might be slow on a local machine (depending on the hardware).
+In any case, local routing requires the route network to be stored on a local computer (we will see how in section \@ref(route-networks)).
+By contrast, remote routing services use a web API to receive queries about origin and destination points, run the routing on the route network on a powerful server and return just the results, the routes, to the user, the so-called client.
 There are many advantages of using remote routing services, including the fact that they can be up-to-date, have global coverage, and run on a set-up that was designed for the job, explaining this section's focus on online routing services.
-Before proceeding, however, you should be aware of the disadvantages of online routing services: they can be slow due to their reliance on internet connections and expensive.
+Before proceeding, however, you should be aware of the disadvantages of online routing services: they are slow or even inaccessible when accessed with a bad internet connection and can be expensive, i.e., not all services are free of charge.
+For instance, the Google routing API sets the limit to 2500 free queries per day ^[`ggmap::route()` lets you use Google's routing API through R.].
 <!-- Todo: add link to Mark's presentation on dodgr vs Google routing costs (RL) -->
 
-A popular and free (for limited uses) routing service is the Open Source Routing Machine ([OSRM](http://project-osrm.org/)).
-Instead of routing *all* desire lines generated in the previous section, which would be time and memory-consuming, we will focus attention on only the desire lines of policy interest.
+A popular and free (for limited uses) routing service is the Open Source Routing Machine ([OSRM](http://project-osrm.org/)).^[Another popular OSM routing API is https://openrouteservice.org/.]
```

Comment:
GIScience Heidelberg is developing https://openrouteservice.org and I think it is not related to OSRM, right?

Reply:
Looks like it's not looking briefly at their source code. It's an OSM routing service thought, thanks for flagging it - looks good.